### DISCUSSION TITLE
About the JS library

### DISCUSSION THREAD

**Discussion by bogdanpetrea** [bhch/django-jsonform#106] [General]
I was wondering why you chose to roll your own JSON form library instead of the more popular https://github.com/rjsf-team/react-jsonschema-form which you mentioned here in an issue.

**Comment by bhch** [comment]
Hi, sorry for the late reply.

I didn't use `react-jsonschema-form` because it didn't have the features that I needed such as file uploads, autocomplete, multiselect, etc.

So I created a pretty basic library with very little features initially. Honestly, I didn't expect it to be used by anyone other than myself. That's why the JavaScript code is a bit of mess.

In retrospect, I think I should have used `react-jsonfshcema-form` with my custom widgets.

I'm planning to do a rewrite now. But that will take some time. I'll see if using `react-jsonschema-form` is a better choice at this point or not.

**Reply by bogdanpetrea** [reply]
> sorry for the late reply.

Oh, don't worry about it... I was just wondering about the decision making process.

For me at least, it's one of the hardest things in development: do you "reinvent" the wheel and tailor it to fit your needs, or do you reuse and extend an existing one that is somewhat good enough? Sometimes it feels like it could be either way, becuase both options come with their own compromises and downsides.

So what you said makes sense, thanks for answering!
